src/Dispatcher.js

```diff
@@ -11,17 +14,49 @@ export default class ReduxDispatcher {
     redux: PropTypes.object.isRequired
   };
 
-  constructor(props) {
-    this.dispatcher = createDispatcher();
-    this.dispatcher.receiveStores(props.stores);
+  getChildContext() {
+    return { redux: this };
   }
 
-  componentWillReceiveProps(nextProps) {
-    this.dispatcher.receiveStores(nextProps.stores);
+  constructor(props, context) {
+    super(props, context);
+
+    this.subscriptions = [];
+    this.emitChange = this.emitChange.bind(this);
+    this.dispatch = this.dispatch.bind(this);
+
+    const initialAtom = dispatch(props.store, undefined, {});
+    this.setAtom(initialAtom);
   }
 
-  getChildContext() {
-    return { redux: this.dispatcher };
+  dispatch(action) {
+    const nextAtom = dispatch(this.props.store, this.atom, action);
+    this.setAtom(nextAtom);
+  }
+
+  setAtom(atom) {
+    this.atom = atom;
+    if (this.state) {
+      this.setState({ atom });
```

Comment:
_Here_ is my only concern with this rewrite. Each `dispatch` call will trigger a `render` of this component. The developer will need to implement a lot of `shouldComponentUpdate` to not lose any perf.